File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Controllers/UnidadeController.cs
using JsonWebToken.Code.Unidade;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace JsonWebToken.Model
{
  public class UnidadeController : ApiController
  { // GET api/<controller>
    public IEnumerable<UnidadeModel> Get()
    {
      UnidadeBLL bll = new UnidadeBLL();
      return bll.SelectUnidade();
    }

    // GET api/<controller>/5
    public string Get(int id)
    {
      // UnidadeBLL bll = new UnidadeBLL();
      return "value";
    }

    // POST api/<controller>
    public void Post([FromBody]UnidadeModel Unidade)
    {
      UnidadeBLL bll = new UnidadeBLL();

      bll.InsertUnidade(Unidade);
    }

    // PUT api/<controller>/5
    public void Put([FromBody]UnidadeModel Unidade, string acao)
    {
      UnidadeBLL bll = new UnidadeBLL();
      if (acao.ToLower().Equals("ativar"))
      {
        bll.AtivarUnidade(Unidade);
      }
      else if (acao.ToLower().Equals("inativar"))
      {
        bll.InativarUnidade(Unidade);
      }

    }

    // DELETE api/<controller>/5
    public void Delete(int id)
    {
      //não realiza esta operacao
      //  UnidadeBLL bll = new UnidadeBLL();

    }
  }
}

File: Backend/Administracao.ConciliacaoAPI/Controllers/UnidadeController.vb
Imports System.Net
Imports System.Net.Http
Imports System.Web.Http
Imports Administracao.ConciliacaoAPI.Code.Unidade
Imports Administracao.ConciliacaoAPI.Model

Namespace Controllers
    Public Class UnidadeController
        Inherits ApiController
        ' GET api/<controller>
        Public Function GetValues() As Object
            Try
                Dim bll As New UnidadeBLL()
                Return New ResultModel(bll.SelectUnidade())
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' GET api/<controller>/5
        Public Function GetValue(id As Integer) As Object
            Try
                Dim bll As New UnidadeBLL()
                Return New ResultModel(bll.SelectUnidadePorId(id))
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' POST api/<controller>
        Public Function PostValue(<FromBody> Unidade As UnidadeModel) As Object
            Try
                Dim bll As New UnidadeBLL()

                bll.InsertUnidade(Unidade)
                Return New ResultModel(Nothing)
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function


        ' PUT api/<controller>/5
        Public Function PutValue(<FromBody> Unidade As UnidadeModel, acao As String) As Object
            Try
                Dim bll As New UnidadeBLL()
                If acao.ToLower().Equals("ativar") Then
                    bll.AtivarUnidade(Unidade)
                ElseIf acao.ToLower().Equals("inativar") Then
                    bll.InativarUnidade(Unidade)
                End If
                Return New ResultModel(Nothing)
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' DELETE api/<controller>/5
        Public Function DeleteValue(id As Integer) As Object
            Try
                'não realiza esta operacao
                '  UnidadeBLL bll = new UnidadeBLL();

                Return New ResultModel(Nothing)
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function
    End Class
End Namespace
